tests/run-make/avr-rjmp-offset/rmake.rs

```diff
@@ -22,7 +22,7 @@ fn main() {
         .input("avr-rjmp-offsets.rs")
         .opt_level("s")
         .panic("abort")
-        .target("avr-unknown-gnu-atmega328")
+        .target("avr-none")
         // normally one links with `avr-gcc`, but this is not available in CI,
         // hence this test diverges from the default behavior to enable linking
         // at all, which is necessary for the test (to resolve the labels). To
```

Comment:
How is this working without specifying a target-cpu?

Reply:
No idea - maybe it's not actually run? (considering the pipelines pass)

Reply:
Does it pass locally? If you can rebase, I'll do one of the try jobs that I think has the lld component.

Or maybe try just `panic!()`ing to see if you can force a failure 😆

Reply:
Actually yeah, I think the default CI job just doesn't do `needs-llvm-components: avr` since LLVM isn't get built. So if you can rebase without fixing the issue, I'll test a different job so we can verify the failure is what we expect.

Reply:
Alright, I don't think this test makes sense anymore - adding `.arg("-Ctarget-cpu=atmega328p")` causes it to crash with `rust-lld: error: unknown emulation: mcu=atmega328p`, but - at the same time - we can't swap `rust-lld` with the actually expected `avr-gcc` linker, because it's not available on the CI:

https://github.com/rust-lang/rust/blob/8c04e395952022a451138dc4dbead6dd6ae65203/tests/run-make/avr-rjmp-offset/rmake.rs#L26

I'd be up for removing it, unless we can get `avr-gcc` on the CI.

Reply:
Ignoring issues with missing symbols, is lld able to link avr binaries correctly if the flags are adjusted? Or was it not working correctly before?

If the former then it would be nicest if this PR could keep it working somehow, so there is less to do once `compiler-builtins` does have the symbols.

Reply:
The issue is that now we're doing `cmd.verbatim_arg(format!("-mmcu={}", codegen_results.crate_info.target_cpu));` within `add_pre_link_args()`, providing `lld` with a flag it doesn't support - and we didn't do that before, because it was a responsibility silently shoved onto `target.json`, which we want to get rid of here, after all.

Considering we have this entire `Linker` abstraction that's available inside `add_pre_link_args()`, I guess we could hack our way around it - e.g. I could add method such as `fn supports_mmcu_flag(&self) -> bool` or whatever, but the only practical use of that would be for this single test (which ain't necessarily a bad thing, ofc.).

Reply:
If somebody wanted to use lld in the future that should theoretically be possible right, assuming the symbols came from somewhere? Meaning it's not like lld doesn't support linking for these targets, just not via the same flags as gcc. Or is using lld for avr broken in some way?

I think we should try to not break lld support if at all possible given it should eventually make using the target easier, but I don't know what the best way to get the flags right is. @workingjubilee do you have any ideas?

Reply:
> If somebody wanted to use lld in the future that should theoretically be possible right, assuming the symbols came from somewhere?

Yes, I think so - it is (should be) just the matter of providing appropriate flags.